src/net/java/sip/communicator/impl/protocol/jabber/ChatRoomJabberImpl.java

```diff
@@ -1028,6 +1028,37 @@ public void sendMessage(Message message)
          }
     }
 
+    /**
+     * Sends the <tt>message</tt> with the json-message extension to the
+     * destination indicated by the <tt>to</tt> contact.
+     *
+     * @param json the json message to be sent.
+     * @throws OperationFailedException if sending the message fails for some
+     * reason.
+     */
+    public void sendJsonMessage(String json)
+            throws OperationFailedException
+    {
+        try
```

Comment:
Can you please avoid the code duplication here?
You can use a private utility method to send a `org.jivesoftware.smack.packet.Message`, and use it in both `ChatRoomJabberImpl#sendMessage(Message)` and `sendJsonMessage(String)`.

The `ChatRoomJabberImpl#sendMessage(Message)` method currently uses the smack `sendMessage(String)` API, but looking at the smack code it should be fine to just replace it with `sendMessage(org.jivesoftware.smack.packet.Message)`.